[DOCS-17] Work with streams in Core (#564)
streamr-dev/streamr-platform: app/src/newdocs/content/streams.mdx

<ScrollableAnchor id="work-with-streams-in-core"><div>

## Work with streams in Core

> Core provides you an easy way to manage your streams. Streams can be created, edited and deleted through our intuitive UI.

Reviewer: ```suggestion
Streamr Core provides you with the tools to create, edit, share and delete your streams as well as view and inspect the streams you have purchased on the Marketplace.
```

### Overview of your streams

> Head out to Streams section to see an overview of your streams.

Reviewer: ```suggestion
Head over to the <Link to={links.userpages.streams}>Streams page</Link> to see an overview of your subscribed streams. The list includes streams you have created and streams that you have purchased on the Marketplace.
```

<img src={streamListing} />

> Stream listing shows statuses of your streams at a glance.

Reviewer: ```suggestion
This stream listing page gives an indication for when data was last pushed to each stream and uses traffic light indicators show the health of each stream, at a glance.
```

> Last received event timestamp will be shown per stream as well as traffic light for your stream status.

Reviewer: ```suggestion

```

| Color        | Status                           |
| ------------ | -------------------------------- |
| Green        | Stream is receiving data         |
| Red          | Stream is not receiving data     |

> You can click the stream name to open the edit view. By clicking the three dots in the last column of the table a stream can be e.g. shared.

Reviewer: ```suggestion
Clicking on a stream will open the stream editor. You may only edit streams that you have created or have explicit permission to edit.

By clicking on the more options icon you may access advanced features for each stream, such as stream sharing and code snippets.
```

> Also a new stream can be created by clicking the blue button in the top right section of the page.

Reviewer: ```suggestion
A new stream can be created by the clicking the "Create Stream" button on the <Link to={links.userpages.streams}>Streams page</Link>.
```

### Editing a stream

> Stream editor provides basic tools for setting up your stream.

Reviewer: ```suggestion
Stream editor provides basic tools for setting up, inspecting and editing your stream.
```

> <img src={streamSettings} />

Reviewer: Looks like there's a screenshot artefact on inside the historical data section - maybe recapture this or reach out to Ross as he's keen to help out for things like this.

Author: Good spot! I recaptured this one.

> #### Details

Reviewer: ```suggestion
#### Details section
```

> Details section is for providing basic information about your stream. Name is used to identify your stream all over the core app.

Reviewer: ```suggestion
The details section provides basic information about your stream. The stream name is used to identify your stream in the Core app. 

The stream description can be used to offer more information about your stream and the kind of data it contains. 

The stream ID is the unique identifier for your stream
and is needed when interacting with the <Link to={links.newdocs.SDKs}>Streamr SDKs</Link>.
```

> Description can be used to tell people more about your stream and what kind of data it contains. Stream ID is an unique identifier for your stream

Reviewer: ```suggestion
```

> and it is used e.g. with SDKs to subscribe and publish data into it.

Reviewer: ```suggestion
```

> #### Configure

Reviewer: ```suggestion
#### Configure section
```

In this section you can configure stream fields and their data types. All fields have a name and a data type.

> Autodetect feature can be used to automatically deduce fields from the stream data.

Reviewer: ```suggestion
The Autodetect feature can be used to automatically deduce fields from the stream's last received message. It is possible to manually correct the data types if automatic detection is not suitable for your stream.
```

It is possible to manually correct the data types if automatic detection is not perfect.

| Setting                                             | Explanation                                                                                                  |
| ----------------------------------------------------| ------------------------------------------------------------------------------------------------------------ |

> | Always try to autoconfigure field names and types   | Streamr will try to automatically configure field names and types as data is published to the stream         |

Reviewer: ```suggestion
| Always try to autoconfigure field names and types   | Streamr will try to automatically configure field names and types on the very first message the stream receives         |
```

| Require all messages in this stream to be signed    | Streamr will not accept messages that are not signed                                                         |

> #### Preview

Reviewer: ```suggestion
#### Preview section
```

This section is used to preview data flowing into the stream. Use the live inspector to make sure your stream is receiving data and it is correctly formatted.

> #### API Access

Reviewer: ```suggestion
#### API Access section
```

> Here you can manage API keys for your streams. An API key which grants access only to this stream can created here with either read or write access.

Reviewer: ```suggestion
Here you can manage API keys for your stream. Keys generated here grant access to only this stream and can have either read or write access. These keys can also be removed if for example, the key has been compromised.
```

Keys can also be deleted which is useful in cases where your key has e.g. leaked.

> #### Historical Data

Reviewer: ```suggestion
#### Historical Data section
```

> In this section historical data for you stream can be managed. You can either add new data to the stream by uploading a CSV file or delete existing data

Reviewer: ```suggestion
This section contains the stream's historical data storage settings and includes tools to add or remove historical data from the stream. 

You can add new data to the stream by uploading a CSV file and you can delete existing historical data from the stream by choosing an inclusive date range. 

The historical data storage period determines how many days your data is retained before it will be removed automatically from the Streamr network. The default period is 365 days.
```
